View3D/Services/MeshOptimizerService.cs

```diff
@@ -103,6 +123,31 @@ public static MeshObject CreatedReducedCopy(MeshObject original, float factor)
             clone.RebuildVertexBuffer();
 
             return clone;
+        }      
+
+        private static MeshObject GetReducedMeshSimplygon(ISimplygon sg, MeshObject originalMesh, float factor)
+        {
+            // -- Copy MeshObject data into Simplygon containers
+            SimplygonHelpers.InitSGGeometryDataObject(sg, out var pGeometryData, originalMesh);
+            SimplygonHelpers.FillSGVertices(ref pGeometryData, originalMesh);
+            SimplygonHelpers.FillSGTriangles(ref pGeometryData, originalMesh);
+            SimplygonHelpers.ReduceSGMesh(sg, pGeometryData, out var newPackedDataSG, factor);
+
+            // -- Make new MeshObject to fit reduces mesh
+            var reducedMesh = originalMesh.Clone(false);
+            reducedMesh.VertexArray = new VertexPositionNormalTextureCustom[newPackedDataSG.GetVertexCount()];
```

Comment:
Here it is possible to make the code simpler and easier to read
reducedMesh.VertexArray =  SimplygonHelpers.GetVertexFromSimpygonMesh(newPackedDataSG)
This way the code structure better document what the intended use is.